experimental/unitsconversion/src/measureunit.rs

```diff
@@ -0,0 +1,243 @@
+            (-3, part)
+        } else if let Some(part) = part.strip_prefix("micro") {
+            (-6, part)
+        } else if let Some(part) = part.strip_prefix("nano") {
+            (-9, part)
+        } else if let Some(part) = part.strip_prefix("pico") {
+            (-12, part)
+        } else if let Some(part) = part.strip_prefix("femto") {
+            (-15, part)
+        } else if let Some(part) = part.strip_prefix("atto") {
+            (-18, part)
+        } else if let Some(part) = part.strip_prefix("zepto") {
+            (-21, part)
+        } else if let Some(part) = part.strip_prefix("yocto") {
+            (-24, part)
+        } else if let Some(part) = part.strip_prefix("ronto") {
+            (-27, part)
+        } else if let Some(part) = part.strip_prefix("quecto") {
+            (-30, part)
+        } else {
+            (0, part)
+        }
+    }
+
+    // TODO: consider returning Option<(i8, &str)> instead of (0, part) for the case when the prefix is not found.
+    // TODO: consider using a trie for the prefixes.
+    // TODO: complete all the cases for the prefixes.
+    /// Get the SI prefix for base 2.
+    /// NOTE:
+    ///     if the prefix is found, the function will return (power, part without the prefix).
+    ///     if the prefix is not found, the function will return (0, part).
+    fn get_si_prefix_base_two(part: &str) -> (i8, &str) {
+        if let Some(part) = part.strip_prefix("kibi") {
+            (10, part)
+        } else if let Some(part) = part.strip_prefix("mebi") {
+            (20, part)
+        } else if let Some(part) = part.strip_prefix("gibi") {
+            (30, part)
+        } else if let Some(part) = part.strip_prefix("tebi") {
+            (40, part)
+        } else if let Some(part) = part.strip_prefix("pebi") {
+            (50, part)
+        } else if let Some(part) = part.strip_prefix("exbi") {
+            (60, part)
+        } else if let Some(part) = part.strip_prefix("zebi") {
+            (70, part)
+        } else if let Some(part) = part.strip_prefix("yobi") {
+            (80, part)
+        } else {
+            (0, part)
+        }
+    }
+
+    /// Get the unit id.
+    /// NOTE:
+    ///    if the unit id is found, the function will return (unit id, part without the unit id and without `-` at the beginning of the remaining part if it exists).
+    ///    if the unit id is not found, the function will return None.
+    fn get_unit_id<'data>(part: &'data str, trie: &ZeroTrie<ZeroVec<'data, u8>>) -> Option<usize> {
+        trie.get(part.as_bytes())
+    }
+
+    /// Process a part of an identifier.
+    /// For example, if the whole identifier is: "square-kilometer-per-second",
+    /// this function will be called for "square-kilometer" with sign (1) and "second" with sign (-1).
+    fn analyze_identifier_part(
+        identifier_part: &str,
+        sign: i8,
+        result: &mut Vec<MeasureUnitItem>,
+        trie: &ZeroTrie<ZeroVec<'_, u8>>,
```

Comment:
Suggestion: A lot of these functions take the trie as a parameter. This suggests that you should consider adding a `MeasureUnitParser` type which has the trie as a field and then make all of these things associated functions with a `&self` parameter to access the trie.

Reply:
good idea, the `MeasureUnitParser` takes the `trie` and the `ConverterFactory` takes the full payload and an instance of `MeasureUnitParser`.

I will add it as a todo to be part of the next PR.